### ISSUE TITLE
Fix S2743 FP: Don't report on static fields which use the type parameter

### ISSUE THREAD

**Issue by hankovich** [SonarSource/sonar-dotnet#5238]
### Repro steps

```csharp
public abstract class Error<TError>
    where TError : Error<TError>, new()
{
    public static readonly TError Instance = CreateDefault(); // S2743: Static fields should not be used in generic types 

    private static TError CreateDefault() => null!;
}
```

### Related information

* C#/VB.NET Plugins version: SonarAnalyzer.CSharp v8.33.0.40503
* Visual Studio version VS 17.1.0 Preview 1.1
* Operating System: Windows

**Comment by pavel-mikula-sonarsource** [comment]
Hi @hankovich,

this is an interesting case where the rule indeed doesn't make sense. I confirm it as a False Positive.

Thank you for reporting it
